### PR TITLE
[VER-98] feat: Add `AddGeneric` component

### PR DESCRIPTION
**Pull request by Ldoppea**
This component is meant to be displayed after clicking on the `Add item` (`+`) buttons and will allow the user to chose a type for the created cipher

When clicking on a type, the user would be redirected to the classic Cipher creation form with the correct type already selected

Clicking some items like `Papers` would not redirect to the classic Cipher creation form but instead would open a new browser tab on the user's Cozy. From here the user will be able to create the item

image

> [!NOTE]
> This PR does not apply the requested design for this specific page, this one will be applied in a future PR that contains the entire redesign for the extension

### CODE REVIEW

File: apps/browser/src/cozy/components/add-generic/add-generic.component.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  async selectType(type: CipherType) {
+    if (type === CipherType.Paper) {
+      await this.cozyClientService.getClientInstance();
+
+      await this.back()
+      // window.open will instantaneously close the popup which may prevent the
+      // router navigation to happens (called inside the `back()` method)
+      // by claling it inside a `setTimeout()` we ensure the call will be done after
```
Comment (zatteo) [inline apps/browser/src/cozy/components/add-generic/add-generic.component.ts:81]: nit: calling

File: apps/browser/src/vault/popup/components/vault/add-edit.component.ts [hunk 2]
```diff
@@ -202,6 +202,24 @@ export class AddEditComponent extends BaseAddEditComponent {
...
+
+    // Cozy customization, override the page title to include the Item's type
+    //*
+    const currentTypeKey = CipherType[this.type]
+
+    this.editMode = this.cipherId != null;
+    if (this.editMode) {
+      this.editMode = true;
+      if (this.cloneMode) {
+        this.cloneMode = true;
```
Comment (zatteo) [inline apps/browser/src/vault/popup/components/vault/add-edit.component.ts:214]: What are the reason for these `this.editMode = true;` and `this.cloneMode = true;` ?

Comment (Ldoppea) [reply apps/browser/src/vault/popup/components/vault/add-edit.component.ts:215]: I'm not sure why. This code is a copy from the parent's component, I only changed the string values.

I added a comment telling that it is copied code here: https://github.com/cozy/cozy-keys-browser/compare/75379a75ac799393dd4042befd514a3858004729..3bf92228d596192e4de0c3584fe625f1f362b09b

File: apps/browser/src/cozy/components/add-generic/add-generic.component.ts (hunk 1 above)
Comment (Ldoppea) [reply apps/browser/src/cozy/components/add-generic/add-generic.component.ts]: Fixed [here](https://github.com/cozy/cozy-keys-browser/compare/75379a75ac799393dd4042befd514a3858004729..3bf92228d596192e4de0c3584fe625f1f362b09b)

File: apps/browser/src/vault/popup/components/vault/add-edit.component.ts (hunk 2 above)
Comment (Ldoppea) [reply apps/browser/src/vault/popup/components/vault/add-edit.component.ts:215]: Note that in the Bitwarden's history there is no hint on why they did this: https://github.com/bitwarden/jslib/commit/337a7ba59f686a5fe8c52d34f3290bbd6ea66669

**Review by zatteo — APPROVED** [review]

**Pull request merged**